ASSETS-59508

igoinggray opened the pull request:
ASSETS-59508: Rights Management Role Enhancement

Summary
- Introduces a new supervisory role for rights management and refactors permission naming from role-based to action-resource pattern for improved clarity and maintainability. This change enables "Manager of Rights Managers" to assign unassigned rights requests to other reviewers.

Problem Statement
- The existing permission system used role-based naming (rights-reviewer, rights-manager) which caused confusion and lacked a clear supervisory hierarchy. Additionally, there was no way for supervisors to assign unassigned requests to specific team members.

Solutio
- New Permission Structure (Action-Resource Pattern):
- manage-rights - Base permission: can review rights requests and self-assign
- admin-rights - Elevated permission: can assign requests to others (automatically includes all manage-rights capabilities)
- Replaces Legacy Permissions:
- ❌ rights-reviewer → ✅ manage-rights
- ❌ rights-manager → ✅ admin-rights
- ❌ Permission aliases (rr, rm) - removed entirely

Changes Made
- Backend (cloudflare/src/api/rightsrequests.js)
- Updated PERMISSIONS constants to new action-resource naming
- Renamed hasRightsReviewerPermission() → hasManageRightsPermission()
- Renamed hasRightsManagerPermission() → hasAdminRightsPermission()
- Bug Fix: Reviewer dropdown now handles domain-level permission inheritance
- Bug Fix: Domain entries (e.g., adobe.com) filtered from dropdown - only individual emails shown
- Updated all endpoint permission checks and documentation
- Backend (cloudflare/src/user.js)
- Removed permissionAliases mapping object
- Removed alias normalization logic (rr → rights-reviewer, rm → rights-manager)
- Permissions now used as-is from configuration sheet
- Frontend (blocks/my-rights-reviews/my-rights-reviews.js)
- Renamed isRightsManager() → isAdminRights()
- Updated to check for 'admin-rights' permission
- Updated button rendering logic for new permission structure
- Reports (blocks/report-rights-requests/)
- Updated column header: RIGHTSREVIEWER → MANAGERIGHTS
- Updated CSV export headers to match
- Updated CSS comments

Key Improvements

- ✅ Permission Hierarchy: admin-rights users automatically inherit all manage-rights capabilities
- ✅ Domain Inheritance: Reviewer dropdown now includes users who inherit permissions from their domain (e.g., users under adobe.com)
- ✅ Cleaner Dropdown: Domain entries (e.g., adobe.com) correctly excluded from assignment dropdown
- ✅ Consistent Naming: All permission checks updated consistently across codebase
- ✅ Better Semantics: Action-resource pattern clearly indicates what users can do, not just who they are

Testing Performed
- API Endpoint Verification
- GET /api/rightsrequests/reviews/reviewersStatus: 200 ✅Reviewer Count: 7 reviewers returned

Verified Behaviors:
- ✅ Users with explicit manage-rights or admin-rights appear in dropdown
- ✅ Users inheriting from domain permissions (e.g., adobe.com → manage-rights) correctly included
- ✅ Domain entries themselves (e.g., adobe.com) correctly excluded

UI Testing
- ✅ Users with manage-rights: See "Assign to Me" button only
- ✅ Users with admin-rights: See both "Assign to Me" and "Assign To..." buttons
- ✅ "Assign To..." dropdown shows all eligible reviewers including domain-inherited permissions
- ✅ Assignment modal functions correctly for both self-assignment and delegation

Code Quality
- ✅ All files pass ESLint with 0 errors
- ✅ No new linter warnings introduced
- ✅ All permission checks updated consistently

Important Notes:
- Users with admin-rights automatically get manage-rights capabilities (no need to list both)
- Domain-level permissions work the same way (e.g., adobe.com,manage-rights)

Steps to Deploy
- Update permissions sheet with new permission names
- Publish changes in Helix Document Authoring
- Users must logout/login to pick up new permissions

Related
JIRA: ASSETS-59508

Checklist
- [x] Code follows project conventions
- [x] All tests pass
- [x] No linter errors
- [x] Documentation updated
- [x] Migration guide provided
- [x] Tested on local environment
- [x] Ready for staging deployment

This PR refactors the permissions system to be more intuitive and extensible while adding the requested supervisory assignment capability. The action-resource pattern makes it clearer what each permission allows and provides a foundation for future permission additions.

Before: https://koassets.adobeaem.workers.dev/my-rights-reviews
After: https://assets-59508-koassets.adobeaem.workers.dev/my-rights-reviews

<img width="1171" height="563" alt="image" src="https://github.com/user-attachments/assets/1d825825-adc1-49cd-bc35-bd18a5840417" />

<img width="957" height="575" alt="image" src="https://github.com/user-attachments/assets/71a920b8-7ad9-4238-9cc5-8be161ca8452" />

igoinggray committed:
ASSETS-59508: Add senior rights reviewer role and assignment functionality

Refactor rights management terminology and introduce new senior-rights-reviewer
role to assign unassigned rights requests to other reviewers.

Backend Changes:
- Add RIGHTS_REVIEWER and SENIOR_RIGHTS_REVIEWER permissions
- Add backwards compatibility for legacy RIGHTS_MANAGER permission
- Create GET /api/rightsrequests/reviews/reviewers endpoint
- Create POST /api/rightsrequests/reviews/assign-to endpoint
- Update existing endpoints to use new permission checks

Frontend Changes:
- Add "Assign To..." button for senior reviewers on unassigned requests
- Create assignment modal with dynamic reviewer dropdown
- Update button text and styling for better UX
- Update reports to use "RIGHTSREVIEWER" terminology

Key Features:
- Senior reviewers can assign requests to other reviewers
- Reviewer list pulled dynamically from permissions config
- Fully backwards compatible with existing rights-manager permission
- Maintains all existing functionality for regular reviewers

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Refactor permissions to rights-reviewer/rights-manager with aliases

Refactor permission naming for improved clarity and intuitive hierarchy.
Replace senior-rights-reviewer with rights-manager for clearer role distinction.

Backend Changes:
- Add permission alias support in user.js (rr → rights-reviewer, rm → rights-manager)
- Update PERMISSIONS constants to use rights-manager for supervisory role
- Rename hasSeniorRightsReviewerPermission to hasRightsManagerPermission
- Remove backwards compatibility for legacy rights-manager permission
- Update all permission checks and API endpoint documentation

Frontend Changes:
- Rename isSeniorRightsReviewer to isRightsManager
- Update permission check to use 'rights-manager'
- Update code comments for new terminology

Permission Structure:
- rights-reviewer (alias: rr) - Base permission for reviewers
- rights-manager (alias: rm) - Supervisory permission for managers

Aliases automatically normalized at login for easier sheet administration.
No breaking changes - clean refactor with improved naming convention.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Fix rights-manager to include reviewer capabilities

Fix critical bug where rights-manager permission did not include
rights-reviewer capabilities, causing 403 errors when managers
tried to view reviews.

Change:
- Update hasRightsReviewerPermission() to recognize rights-manager
  as including all rights-reviewer capabilities
- Rights managers now only need 'rm' permission, not both 'rm' and 'rr'

This makes rights-manager a proper superset of rights-reviewer,
where managers can do everything reviewers can do plus assign to others.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Add debug logging for permission troubleshooting

Add temporary console logging to diagnose why Assign To button
appears for users with only rights-reviewer permission.

Logs:
- User permissions array
- isRightsManager() result
- Full user object

This is temporary debug code to be removed after issue is resolved.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Add debug logging for permission sheet loading

Add temporary server-side console logging to diagnose why
rights-reviewer from sheet appears as rights-manager in session.

Logs show:
- Wildcard, domain, and email permissions from sheet
- Raw permissions before alias normalization
- Final permissions after alias normalization

This is temporary debug code to be removed after issue is resolved.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Add temporary debug endpoint for permissions

Add /api/debug/permissions endpoint to check what server reads
from the permissions sheet in real-time.

This will help diagnose why rights-reviewer appears as rights-manager
in the session.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Add test mode for permission alias verification

Add TEST_PERMISSION cookie support to override sheet permissions
for isolated testing of alias normalization.

Enables testing of:
- rr → rights-reviewer
- rights-reviewer → rights-reviewer
- rm → rights-manager
- rights-manager → rights-manager

Also adds /api/debug/test-alias endpoint for quick alias checks.

This is temporary debug code to be removed after testing.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Fix reviewer dropdown to only show actual reviewers

Fixed listAvailableReviewers endpoint to:
1. Normalize permission aliases (rr, rm) before filtering
2. Remove hardcoded RIGHTS_REVIEWERS fallback list

Now dropdown only shows users with rights-reviewer or rights-manager
permissions from the sheet, properly handling aliases.

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
ASSETS-59508: Remove debug logging and test code - production ready

Removed all temporary debugging code:
- Removed TEST_PERMISSION override in user.js
- Removed debug console.log statements in user.js
- Removed debug endpoints from index.js (/api/debug/permissions, /api/debug/test-alias)
- Removed debug console.log statements in my-rights-reviews.js

Code is now production-ready with:
✅ Permission alias support (rr → rights-reviewer, rm → rights-manager)
✅ Reviewer dropdown properly filtered to only show actual reviewers
✅ All functionality tested and working
✅ Clean, production-ready code

JIRA: https://jira.corp.adobe.com/browse/ASSETS-59508

igoinggray committed:
committed index.js to remove debug statements.  Removed markdown documents accidentially committed.

jfait commented on blocks/report-rights-requests/report-rights-requests.js:
```diff
@@ -1126,7 +1126,7 @@ function exportToCSV() {
     'TITLE',
     'RIGHTSREQUESTSTATUS',
-    'RIGHTSMANAGER',
+    'RIGHTSREVIEWER',
```
<img width="1544" height="184" alt="image" src="https://github.com/user-attachments/assets/bb8bfb90-fa6c-4107-ac06-cb3f9c8befaa" />

jfait replied:
I get my confusion now. we're change role name/definition in this PR.


>Introduce a new role for a Manager of Rights Manager

the person doing the reviews is defined already as a RightManager.

So what we had before was a rights manager.
However now we have renamed an existing role to take it over for a new role.

@inedoviesov1 , what is the name of this new role? Does TCCC really refer to them as rights manager manager?
?

do they want us to change the existing meaning of right-manager or will that confuse them?

igoinggray committed:
feat(ASSETS-59508): Refactor permissions to action-resource pattern

Introduce new permission structure with action-resource naming:
- manage-rights (base): can review and self-assign rights requests
- admin-rights (elevated): can assign to others (includes manage-rights)

Replace legacy permissions (rights-reviewer, rights-manager) with
clearer action-resource pattern for improved clarity and maintainability.

Changes:
- Backend: Update PERMISSIONS constants and helper functions
- Backend: Remove permission alias normalization (rr/rm)
- Backend: Fix reviewer dropdown to handle domain inheritance
- Backend: Filter domain entries from reviewer dropdown
- Frontend: Update permission checks (isAdminRights)
- Reports: Update column headers (RIGHTSREVIEWER → MANAGERIGHTS)

Key improvements:
- admin-rights users automatically inherit manage-rights capabilities
- Reviewer dropdown now includes users who inherit from domain permissions
- Domain entries (e.g., adobe.com) correctly excluded from dropdown
- All permission checks updated consistently across codebase

Migration: Sheet admins should update permissions sheet to use
manage-rights and admin-rights instead of old permission names.

Tested: All scenarios verified including permission hierarchy,
domain inheritance, dropdown filtering, and API endpoints.

igoinggray committed:
fix(ASSETS-59508): Add delay for Cloudflare KV propagation after assignment

Fix race condition where UI refresh happened before KV write propagated,
causing tab counts and review lists to show stale data after assignment.

Changes:
- Add 800ms delay after 'Assign to Me' before refreshing reviews
- Add 800ms delay after 'Assign To...' before refreshing reviews
- Ensures Cloudflare KV global propagation completes before data refresh
- Eliminates need for logout/login to see correct assignment state

Fixes issue where reviews disappeared from Unassigned but didn't appear
in My Reviews until session refresh. Tab counts now update correctly
and users stay on current tab for continued workflow.

igoinggray committed:
fix(ASSETS-59508): Fix linter errors in KV propagation delay code

Fix trailing spaces and arrow function syntax to pass ESLint:
- Add parentheses around arrow function parameters
- Use statement block syntax to avoid executor return value
- Remove trailing whitespace

All linter errors resolved, 0 errors remaining.

igoinggray commented on local.sh:
```diff
@@ -68,6 +68,13 @@ if nc -z "localhost" "8787" > /dev/null 2>&1; then
   exit 1
 fi
 
+# Initial React build to ensure fresh artifacts
```
I didn't mean to check this in.  This was added for my purposes because I wouldn't always seem my changes when running just ./local.sh, which runs npm run dev.  This guaranteed.everything was loaded properly.  I'll remove change.

igoinggray commented on cloudflare/src/api/rightsrequests.js:
```diff
@@ -72,9 +77,15 @@ export async function rightsRequestsApi(request, env) {
   if (request.method === 'GET' && path.endsWith('/rightsrequests/reviews')) {
     return listReviewsForReviewer(request, env);
   }
+  if (request.method === 'POST' && path.endsWith('/rightsrequests/reviews/assign-to')) {
```
I can make this change.

igoinggray committed:
refactor: consolidate assignment API into single unified endpoint

Merge self-assignment and assign-to-other into /api/rightsrequests/reviews/assign
with optional assigneeEmail parameter. Remove duplicate assignReviewToReviewer()
function. Update frontend to use unified endpoint.

Also restore local.sh to original state.

JIRA: ASSETS-59508

igoinggray committed:
refactor: consolidate assignment API endpoints; restore local.sh

Consolidate assignment APIs:
- Merge /api/rightsrequests/reviews/assign and /assign-to into single endpoint
- Updated assignReview() to accept optional assigneeEmail parameter
  - Self-assignment (no assigneeEmail): requires manage-rights or admin-rights
  - Assign to other (assigneeEmail provided): requires admin-rights
- Removed assignReviewToReviewer() function and /assign-to route handler
- Updated frontend to call unified /assign endpoint
- Validates assignee permissions including domain inheritance
- Updated all comments to use manage-rights/admin-rights terminology

Other changes:
- Restore local.sh to original state (reverting accidental changes)

JIRA: ASSETS-59508

igoinggray committed:
fix: improve button hover states for better readability

Replace harsh hover color changes with opacity + shadow effects on all
action buttons. Ensures text remains readable while maintaining clear
visual feedback.

- Filled buttons: opacity 0.9 + shadow + subtle lift
- Outlined buttons: light tint + darker border + shadow

JIRA: ASSETS-59508

igoinggray committed:
fix: improve button hover states for better readability

igoinggray committed:
refactor: rename reports-admin permission to admin-reports
